cmd/agent/api/agent/agent.go

```diff
@@ -179,6 +191,64 @@ func getStatus(w http.ResponseWriter, r *http.Request) {
 	w.Write(jsonStats)
 }
 
+func streamLogs(w http.ResponseWriter, r *http.Request) {
+	log.Info("Got a request for stream logs.")
+	w.Header().Set("Transfer-Encoding", "chunked")
+	logMessageReceiver := logs.GetMessageReceiver()
+	if logMessageReceiver == nil {
+		http.Error(w, "The logs agent is not running", 405)
+		w.(http.Flusher).Flush()
```

Comment:
It might be worth asserting on the safety of the typecasts to `http.Flusher` and `http.CloseNotifier`? I fear that if at some point in the future we add a middleware in the agent we might accidentally break the assumption that `http.ResponseWriter` implements these interfaces.